static/app/components/replays/replayContext.tsx

```diff
@@ -420,8 +420,11 @@ export function Provider({children, replay, initialTimeOffset = 0, value = {}}:
   }, []);
 
   // Only on pageload: set the initial playback timestamp
-  // Don't include `setCurrentTime` in the hook deps array because it changes
+  // Do not include `setCurrentTime` in the hook deps array because it changes
   // on each play/pause state change.
+  // Do include replayerRef.current in the hook deps. The rule warns that since
+  // it is a ref changing it will not cause a re-render. However, when that
+  // value changes we will get a re-render because other state values have changed.
```

Comment:
If it's being re-render because other state values have changed.... shouldn't those state values be in the hook deps?

Reply:
`replayerRef.current` changes based on some conditions, so i didn't want to list out all those variables because then the hook would re-fire and i'd have to check `replayerRef.current`...

maybe the answer is that instead of `useRef` i need `useState` now.